### DISCUSSION TITLE
How do I correctly import plugins in stable tauri?

### DISCUSSION THREAD

**Discussion by ikripaka** [tauri-apps/tauri#11411] [Q&A] [answered]
### Description
Now I'm fully frustrated about importing plugins in stable tauri version. I have another project on it, currently switching from tauri-v2.0.0-beta.19 to tauri-v2.0.4. Now I faced with problem of giving correct permissions for frontend. 
Currently I have such problems: 
*  `Unhandled Promise Rejection: event.listen not allowed. Permissions associated with this command: event:allow-listen, event:default`
* `dialog.open not allowed. Permissions associated with this command: dialog:allow-open, dialog:default`
Ok, I understand that I have to add this permissions to my `capabilities/main.json` file, so here it is: 
```json
{
  "$schema": "../gen/schemas/desktop-schema.json",
  "identifier": "main-capability",
  "description": "Capability for the main window",
  "windows": [
    "main"
  ],
  "permissions": [
    "core:default",
    "window:default",
    "path:default",
    "event:default",
    "event:allow-listen",
    "app:default",
    "resources:default",
    "menu:default",
    "tray:default",
    "fs:default",
    "fs:write-all",
    "fs:read-all",
    "shell:allow-open",
    "dialog:allow-save",
    "dialog:allow-open"
  ],
  "capabilities": []
}
```

tauri.conf.json:
```json
{
  "productName": "DesktopA",
  "identifier": "app-pc",
  "build": {
    "beforeDevCommand": "npm run dev",
    "devUrl": "http://localhost:1420",
    "beforeBuildCommand": "npm run build",
    "frontendDist": "../dist"
  },
  "app": {
    "windows": [
      {
        "fullscreen": false,
        "height": 720,
        "minHeight": 600,
        "minWidth": 845,
        "resizable": true,
        "title": "Desktop app",
        "width": 845
      }
    ],
    "security": {
      "csp": null
    }
  },
  "bundle": {
    "active": true,
    "category": "Utility",
    "copyright": "Amazooon llc LLC, all rights reserved",
    "icon": [
      "icons/32x32.png",
      "icons/128x128.png",
      "icons/128x128@2x.png",
      "icons/icon.icns",
      "icons/icon.ico"
    ],
    "publisher": "Amazooon llc",
    "shortDescription": "app for pc",
    "targets": [
      "deb",
      "msi"
    ],
    "linux": {
      "deb": {
        "files": {
          "file": "example_file.so"
        }
      }
    },
    "windows": {
      "wix": {
        "language": ["en-US","uk-UA"]
      }
    }
  },
  "plugins": {
    "fs": {
      "requireLiteralLeadingDot": false
    }
  }
}
```
Needless to note, in the main.rs file before running, as usual I add plugins for usage: 
```rust
    .plugin(tauri_plugin_shell::init())
    .plugin(tauri_plugin_dialog::init())
    .plugin(tauri_plugin_fs::init())
```
### Tauri info
```
[✔] Environment
    - OS: Ubuntu 22.4.0 x86_64 (X64)
    ✔ webkit2gtk-4.1: 2.44.3
    ✔ rsvg2: 2.52.5
    ✔ rustc: 1.81.0 (eeb90cda1 2024-09-04)
    ✔ cargo: 1.81.0 (2dbb1af80 2024-08-20)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-x86_64-unknown-linux-gnu (default)
    - node: 20.11.1
    - pnpm: 8.15.4
    - npm: 10.9.0

[-] Packages
    - tauri 🦀: 2.0.4
    - tauri-build 🦀: 2.0.1
    - wry 🦀: 0.46.1
    - tao 🦀: 0.30.3
    - tauri-cli 🦀: 2.0.3

[-] Plugins
    - tauri-plugin-shell 🦀: 2.0.1
    - tauri-plugin-dialog 🦀: 2.0.1
    - tauri-plugin-fs 🦀: 2.0.1

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1420/
```

### Reproduce
Actually, I can't reproduce that..., and all posts mentioned before with `fs`, `dialog` problems doesn't help me at all.
* https://github.com/tauri-apps/tauri/discussions/10726
* https://github.com/tauri-apps/tauri/issues/8967

Maybe you can give me any advice on how to fix that? In advance, thanks for help!!

**Comment by FabianLars** [comment]
Can you try it in a fresh create-tauri-app and compare? If it has the same problems, please upload it to a repo for us to check out.

Also, permissions for the non-plugin api should be prefixed with `core:` like `core:event:default`. idk why the error doesn't mention it. iirc it does though so maybe some package was outdated when you saw that error?
Of course that doesn't explain why the plugins don't work so i hope you'll be able to reproduce this somehow.

**Reply by ikripaka** [reply, marked as answer]
Sorry, but I can't reproduce that bug( All projects in which I wanted to show the bug, just work! 
Needless to say, I managed somehow to correctly build it by copying all my files into the new project, created with `cargo create-tauri-app`. I don't know how, but it works like a magic wand :) I changed only one file with capabilities, as permissions were changed.
Yes, I understand that nothing can happen without some interference, so I believe that beta and stable versions differ somehow which makes the transition difficult. 

The main recipe: 
* create a project with `cargo create-tauri-app` and move your files into another one.

**Reply by ghost** [reply]
I have been trying for 1-2 hours. I tried every method, I tried your method but it did not work. This error indicates problems with the API configuration and I am going crazy.

**Reply by ghost** [reply]
Even though I allowed everything, it gives an error. I wanted to define the builder in the code block over another structure, but it was illogical because the basic functions are already on the` default`.
